utilities/tools/Clear-ManagementGroupDeployment.ps1

```diff
@@ -0,0 +1,96 @@
+﻿
+<#
+.SYNOPSIS
+Bulk delete all deployments on the given management group scope
+
+.DESCRIPTION
+Bulk delete all deployments on the given management group scope
+
+.PARAMETER ManagementGroupId
+Mandatory. The Resource ID of the Management Group to remove the deployments for.
+
+.PARAMETER DeploymentStatusToExclude
+Optional. The status to exlude from removals. By default, we exclude any deployment that is in state 'running'.
+
+.EXAMPLE
+Clear-ManagementGroupDeployment -ManagementGroupId 'MyManagementGroupId'
+
+Bulk remove all 'non-running' deployments from the Management Group with ID 'MyManagementGroupId'
+#>
+function Clear-ManagementGroupDeployment {
+
+
+    [CmdletBinding(SupportsShouldProcess)]
+    param (
+        [Parameter(Mandatory = $true)]
+        [string] $ManagementGroupId,
+
+        [Parameter(Mandatory = $false)]
+        [string] $DeploymentStatusToExclude = 'running'
+    )
+
+    # Load used functions
+    . (Join-Path $PSScriptRoot 'helper' 'Split-Array.ps1')
+
+    $getInputObject = @{
+        Method  = 'GET'
+        Uri     = "https://management.azure.com/providers/Microsoft.Management/managementGroups/$ManagementGroupId/providers/Microsoft.Resources/deployments/?api-version=2021-04-01"
+        Headers = @{
+            Authorization = 'Bearer {0}' -f (Get-AzAccessToken).Token
+        }
+    }
+    $response = Invoke-RestMethod @getInputObject
```

Comment:
Why not using `Invoke-AzRestMethod`? Cleaner code avoiding Headers and consistent with what we use for removal.

Reply:
I tried - and failed. It returns error messages above subscriptions not being set - which is ironic as I want to execute actions on an above subscription-level. Troubleshooting didn't return and results so I went with the general and working `Invoke-RestMethod`.

Reply:
disappointing but fine :)